### PR TITLE
UI tests for header

### PR DESCRIPTION
**Pull request by Erin007**
UI tests to check that the correct links are in the header for teachers, students and signed out users in English and Spanish.

### CODE REVIEW

**Comment by Erin007** [comment]
cc: @tanyaparker

File: dashboard/test/ui/features/header.feature [hunk 1]
```diff
@@ -0,0 +1,65 @@
+@no_mobile
+Feature: Header navigation bar
+
+Scenario: Signed out user in English should see 6 header links
+  Given I am on "http://code.org/"
+  And I dismiss the language selector
+  And I wait to see "#headerlinks"
+  And I wait to see ".headerlink"
```
Comment (breville) [inline dashboard/test/ui/features/header.feature]: Once you've waited for one indicator that everything has loaded, I think it's more readable to just race through everything that you know will be there.  I'd just use waits when it's actually necessary.

Comment (breville) [reply dashboard/test/ui/features/header.feature]: And in the case where we just have static HTML, I don't think we need to check for both `#headerlinks` and `.headerlink`.

File: dashboard/test/ui/features/header.feature [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  And I wait until ".headerlink" contains text "Sections"
+  And I wait until ".headerlink" contains text "Professional Learning"
+
+Scenario: Signed out user in Spanish should see 2 header links
+  Given I am on "http://code.org/lang/es"
+  Then check that I am on "http://code.org/"
+  And I dismiss the language selector
+  And I wait to see "#headerlinks"
+  And I wait to see ".headerlink"
+  And I wait until ".headerlink" contains text "Courses"
```
Comment (breville) [inline dashboard/test/ui/features/header.feature]: Look up "text from key" in steps.rb to see how we can look up non-en strings.

**Pull request merged**